api/src/main/java/org/apache/brooklyn/api/catalog/CatalogConfig.java

```diff
@@ -34,5 +34,7 @@
      * a higher value appears higher in the list. the default is 1.
      * (negative values may be used to indicate advanced config which might not be shown unless requested.) */ 
     double priority() default 1;
-    
+
+     /** a pinned configuration means that it should be displayed, regardless of its priority */
+    boolean pinned() default false;
```

Comment:
I would rather modify https://github.com/apache/brooklyn-library/pull/65 to add `pinned = true` to all annotated field, than changing this to `true`. Mark by default annotated fields as pinned feels wrong